homeassistant/components/matter/button.py

```diff
@@ -0,0 +1,149 @@
+"""Matter Button platform."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+from chip.clusters import Objects as clusters
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity, MatterEntityDescription
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter Button platform."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.BUTTON, async_add_entities)
+
+
+@dataclass(frozen=True)
+class MatterButtonEntityDescription(ButtonEntityDescription, MatterEntityDescription):
+    """Describe Matter Button entities."""
+
+    command: Callable[[], Any] | None = None
```

Comment:
Why is the command optional and then asserted to be not None below? It should not be optional, I think.